ClosedXML/Excel/Ranges/XLRangeColumnComparer.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace ClosedXML.Excel
+{
+    internal class XLRangeColumnComparer : IComparer<IXLRangeColumn>
+    {
+        private readonly IXLSortElements _sortElements;
+
+        internal XLRangeColumnComparer(IXLSortElements sortElements)
+        {
+            this._sortElements = sortElements;
+        }
+
+        public int Compare(IXLRangeColumn x, IXLRangeColumn y)
+        {
+            foreach (IXLSortElement e in _sortElements)
+            {
+                var thisCell = (XLCell)x.Cell(e.ElementNumber);
+                var otherCell = (XLCell)y.Cell(e.ElementNumber);
+                int comparison;
+                bool thisCellIsBlank = thisCell.IsEmpty();
+                bool otherCellIsBlank = otherCell.IsEmpty();
+                if (e.IgnoreBlanks && (thisCellIsBlank || otherCellIsBlank))
+                {
+                    if (thisCellIsBlank && otherCellIsBlank)
+                        comparison = 0;
+                    else
+                    {
+                        if (thisCellIsBlank)
+                            comparison = e.SortOrder == XLSortOrder.Ascending ? 1 : -1;
+                        else
+                            comparison = e.SortOrder == XLSortOrder.Ascending ? -1 : 1;
+                    }
+                }
+                else
+                {
+                    if (thisCell.DataType == otherCell.DataType)
+                    {
+                        if (thisCell.DataType == XLDataType.Text)
+                        {
+                            comparison = e.MatchCase
+                                             ? thisCell.InnerText.CompareTo(otherCell.InnerText)
+                                             : String.Compare(thisCell.InnerText, otherCell.InnerText, true);
+                        }
```

Comment:
Yes, I think this requires a separate PR, to be done first before continuing this PR.

Reply:
It can go after too, because may also require adjusting formulas on cells movement (which you implemented in this PR).

Reply:
Ok, let's park it. 

In fact, I dislike the `InnerText` property and `string _cellValue` field. I don't really see its purpose. I think using `.Value` or `.CachedValue` should cover all cases. Or `.GetString()` as you propose. And I think `string _cellValue` should be replaced with `object _cellValue` to avoid the conversion to `string` values in `XLCell.SetKnownTypedValue`.